### DISCUSSION TITLE
Mindstorms EV3 (Micro Python) - Motor.run() not working

### DISCUSSION THREAD

**Discussion by DylanBPY** [micropython/micropython#13029] [Other] [answered]
This was also posted on Stack Overflow, it has been 2 days and I have no answers. I am wondering if somebody here can help me.

I have a Lego Mindstorms EV3 running MicroPython. 
When I try to use this code, it does not work:
```python
#!/usr/bin/env pybricks-micropython
from pybricks.robotics import DriveBase, Stop
from pybricks.hubs import EV3Brick
from pybricks.ev3devices import Motor, InfraredSensor
from pybricks.parameters import Port, Direction
from pybricks.tools import wait

# Initialize the EV3 Brick.
ev3 = EV3Brick()

# Initialize Wheels
RightWheel = Motor(Port.B)
LeftWheel = Motor(Port.C)
Robot = DriveBase(LeftWheel, RightWheel, 275.2, 165)
Robot.settings(straight_speed=1000)

# Initialize Motors and Sensors
blade_motor = Motor(Port.D, positive_direction=Direction.CLOCKWISE, gears=None)
infrared_sensor = InfraredSensor(Port.S1)
pressed = infrared_sensor.keypad()

# Play a sound to tell us when we are ready to start moving
ev3.speaker.beep()

LeftWheel.run(1000)
```
If I replace ```python LeftWheel.run(1000)``` with ```python LeftWheel.run_time(1000, 5000)``` It works perfectly.

I have no idea why this would happen, I have looked everywhere, and it seems to be just me that is having this issue.

**Comment by DylanBPY** [comment]
Has to be in while loop

**Reply by dlech** [reply, marked as answer]
FYI, the best place to ask for help with Pybricks is https://github.com/orgs/pybricks/discussions.

Technically, it doesn't have to be in a while loop. You just need something that makes sure that the program does not reach the end. Motors are stopped when the program reaches the end. `run_time()` works because it blocks program execution for some amount of time. But `run()` does not block program execution, so the program ends immediately and stops the motors.

**Reply by DylanBPY** [reply]
Thank you for clarifying that.
